Assets/Scripts/Combat/Combantants/Combatant.cs

```diff
@@ -18,24 +18,31 @@ public abstract class Combatant : MonoBehaviour
     GameObject HealthBarUIPanel;
 
     private GameObject HealthBar;
-    private bool displayHealthBar;
 
     public bool isAlive = true;
+    private bool inCombat;
 
     public void Start()
     {
+        inCombat = false;
+        HealthBar = Instantiate(HealthBarPrefab);
+        HealthBar.transform.SetParent(HealthBarUIPanel.transform);
+        HealthBar.SetActive(false);
+    }
+
+    public void ShowHealth()
+    {
+        inCombat = true;
         CombatSystem = GameObject.FindGameObjectWithTag("CombatSystem").GetComponent<CombatSystem>();
 
         CurrentHealthPoints = MaxHealthPoints;
```

Comment:
I'm not sure why `ShowHealth` is doing this initialization job for the combat. It may be the case right now that ShowHealth is the first function called when a combat is started, but I think it would be more elegant if there was a proper `EnterCombat` method that took care of 1. Initialization and 2. Showing health, such that `ShowHealth` truly only shows health (and has no other effect).